src/components/preference-item/preference-item.component.spec.ts

```diff
@@ -1,37 +1,112 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { PreferenceItemComponent } from './preference-item.component';
 import { BehaviorSubject } from 'rxjs';
+import { Preference, PreferenceWithMeta } from '../../services/preference/types';
+import { MatCardModule } from '@angular/material/card';
+
+interface TestPreference extends Preference {
+  num: number;
+  bool: boolean;
+  select: string;
+}
 
 describe('PreferenceItemComponent', () => {
-  let fixture: ComponentFixture<PreferenceItemComponent<any>>;
-  let component: PreferenceItemComponent<any>;
+  let boolFixture: ComponentFixture<PreferenceItemComponent<TestPreference, 'bool'>>;
+  let numFixture: ComponentFixture<PreferenceItemComponent<TestPreference, 'num'>>;
+  let selectFixture: ComponentFixture<PreferenceItemComponent<TestPreference, 'select'>>;
+  let boolComponent: PreferenceItemComponent<TestPreference, 'bool'>;
+  let numComponent: PreferenceItemComponent<TestPreference, 'num'>;
+  let selectComponent: PreferenceItemComponent<TestPreference, 'select'>;
+  let preference$: BehaviorSubject<PreferenceWithMeta<TestPreference>>;
 
   beforeEach(() => {
     TestBed.configureTestingModule({
+      imports: [
+        MatCardModule,
+      ],
       declarations: [
         PreferenceItemComponent,
       ],
     });
-    fixture = TestBed.createComponent(PreferenceItemComponent);
-    component = fixture.componentInstance;
+    boolFixture = TestBed.createComponent<PreferenceItemComponent<TestPreference, 'bool'>>(PreferenceItemComponent);
+    numFixture = TestBed.createComponent<PreferenceItemComponent<TestPreference, 'num'>>(PreferenceItemComponent);
+    selectFixture = TestBed.createComponent<PreferenceItemComponent<TestPreference, 'select'>>(PreferenceItemComponent);
+    boolComponent = boolFixture.componentInstance;
+    numComponent = numFixture.componentInstance;
+    selectComponent = selectFixture.componentInstance;
+
+    preference$ = new BehaviorSubject<PreferenceWithMeta<TestPreference>>({
+      enabled: true,
+      num: 0,
+      bool: true,
+      select: 'option1',
+      _meta: {
+        enabled: {
+          type: 'boolean',
+          defaultValue: true,
+        },
+        num: {
+          type: 'number',
+          defaultValue: 0,
+        },
+        bool: {
+          type: 'boolean',
+          defaultValue: true,
+        },
+        select: {
+          type: 'select',
+          defaultValue: 'option1',
+          options: ['option1', 'option2'],
+        }
+      },
```

Comment:
Just to reiterate my comment from the Preference types doc below, I think Preferences would be easier to read without the `_meta` being stored in a distinct object separate from the current value.